templates/server/src/main/resources/archetype-resources/core/src/main/java/__packageInPathFormat__/general/common/impl/security/BaseUserDetailsService.java

```diff
@@ -79,14 +80,64 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
     return authorities;
   }
 
-  private Collection<String> getRoles(String username) {
+  /**
+   * Lists roles of AWS from IAM profile of pointed user.
+   *
+   * @param userName        given userName in AWS
+   * @param showErrorOutput flag marking outputing errors to console
+   * @return collection with user's roles
+   */
+  public static Collection<String> obtainRoles(String userName, boolean showErrorOutput) {
+    Collection<String> roles = new ArrayList<String>();
+
+    final String baseCommand = "aws iam list-groups-for-user --user-name";
+    String command = String.format("%s %s", baseCommand, userName);
+
+    BufferedReader reader = null;
+    Process process;
+    try {
+      process = Runtime.getRuntime().exec(command);
+      process.waitFor();
+      reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+
+      String line;
+      while ((line = reader.readLine()) != null) {
+        String theLine = line.trim();
+        if (theLine.startsWith("\"GroupName\": ")) {
+          String roleName = theLine.substring(14, theLine.length() - 1);
+          roles.add(roleName);
+        }
+      }
+    } catch (Exception exc_1) {
+      if (showErrorOutput) {
+        exc_1.printStackTrace();
+      }
+    } finally {
+      if (reader != null) {
+        try {
+          reader.close();
+        } catch (Exception exc_2) {
```

Comment:
you could use try-with-resource syntax to reduce your code and make it simpler and more compact.
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html